### ISSUE TITLE
HIR cleanup: Unify generic param bounds and where clause

### ISSUE THREAD

**Issue by jdupak** [Rust-GCC/gccrs#2724]
This is syntax sugar with no difference in meaning. No need to have tham split in two places.

@philberty agree?

**Comment by CohenArthur** [comment]
sounds like a good idea to me

**Comment by philberty** [comment]
Yes this makes me happy @jdupak go for it. I left them because I was not sure if there was special handling here
